webapp/src/Migrations/Version20210404121354.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20210404121354 extends AbstractMigration
+{
+    public function getDescription() : string
+    {
+        return 'Use an auto increment integer column as primary key for judgehosts.';
+    }
+
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        // First, add the new field on the judgehost table. For this to work, we first need to add a unique index on hostname, since otherwise foreign keys will break
+        $this->addSql('CREATE UNIQUE INDEX hostname ON judgehost (hostname)');
+        $this->addSql('ALTER TABLE judgehost DROP PRIMARY KEY');
+        $this->addSql('ALTER TABLE judgehost ADD judgehostid INT UNSIGNED AUTO_INCREMENT NOT NULL COMMENT \'Judgehost ID\' FIRST, ADD PRIMARY KEY (judgehostid)');
+
+        // Now add the new judgehostid field on judgetasks and judgings, including indices
+        $this->addSql('DROP INDEX hostname_valid_priority ON judgetask');
+        $this->addSql('DROP INDEX hostname_jobid ON judgetask');
+        $this->addSql('DROP INDEX specific_type ON judgetask');
```

Comment:
Now we have two sequential migrations in master that create big indices only to immediately drop them. We squash commits for more clarity - what are the thoughts about squashing migrations?

Reply:
I am fine with this if others are ok with manually fixing their db (probably just recreating it)

Reply:
Should be done